Do not shrink artifacts.xml during processsing

```
java.vm.version=25.172-b11
java.vm.name=Java HotSpot(TM) 64-Bit Server VM
java.vm.specification.vendor=Oracle Corporation
java.vm.specification.version=1.8
java.vm.vendor=Oracle Corporation

OS name: "windows 8.1", version: "6.3", arch: "amd64", family: "dos"

```

After creating p2 proxy to "http://download.eclipse.org/releases/mars/" and attempt to use it in the local maven build it fails with error like:
```
[Fatal Error] :112850:13: XML document structures must start and end within the same entity.
[ERROR] Internal error: java.lang.RuntimeException: org.eclipse.equinox.p2.core.ProvisionException: Unable to read repository at https://.../repository/eclipse-mars. Unable to read repository at file:/C:/Users/.../.m2/repository/.cache/tycho/p2-repository-metadata/2116131570. XML document structures must start and end within the same entity. -> [Help 1]
```

The mentioned file "2116131570" appears to be zip with 1 file named "2116131570" inside it.

The inner file "2116131570" was an XML that ends with
```
      <property name="format" value="packed"></property>
      <pro
```
So, it was shrinked somethere during processing.

The solution is to work with XML processing more carefully, thanks to @afischer211 who found it. PR is following.

--- a/CONTRIBUTORS.md
+++ b/CONTRIBUTORS.md
@@ -24,4 +24,7 @@ Sonatype internal people:
 
 External contributors:
 
+* [@afischer211](https://github.com/afischer211/) (Alexander Fischer)
+* [@ruspl-afed](https://github.com/ruspl-afed/) (Alexander Fedorov)
+
 ![Possibly You!](http://i.imgur.com/A3eScYul.jpg)

--- a/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java
+++ b/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java
@@ -92,6 +92,7 @@ public class ArtifactsXmlAbsoluteUrlRemover
             XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();
             XMLEventReader reader = null;
             XMLEventWriter writer = null;
+            //try-with-resources will be better here, but XMLEventReader and XMLEventWriter are not AutoCloseable
             try {
               reader = inputFactory.createXMLEventReader(xmlIn);
               writer = outputFactory.createXMLEventWriter(xmlOut);

--- a/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java
+++ b/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/ArtifactsXmlAbsoluteUrlRemover.java
@@ -90,10 +90,21 @@ public class ArtifactsXmlAbsoluteUrlRemover
           try (OutputStream xmlOut = xmlOutputStream(extension, tempFile)) {
             XMLInputFactory inputFactory = XMLInputFactory.newFactory();
             XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();
-            XMLEventReader reader = inputFactory.createXMLEventReader(xmlIn);
-            XMLEventWriter writer = outputFactory.createXMLEventWriter(xmlOut);
-
-            streamXmlToWriterAndRemoveAbsoluteUrls(reader, writer);
+            XMLEventReader reader = null;
+            XMLEventWriter writer = null;
+            try {
+              reader = inputFactory.createXMLEventReader(xmlIn);
+              writer = outputFactory.createXMLEventWriter(xmlOut);
+              streamXmlToWriterAndRemoveAbsoluteUrls(reader, writer);
+              writer.flush();
+            } finally {
+              if (reader != null) {
+                reader.close();
+              }
+              if (writer != null) {
+                writer.close();
+              }
+            }
           }
           return convertFileToTempBlob(tempFile, repository);
         }

